solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -383,6 +388,7 @@ public CoreContainer(NodeConfig config, CoresLocator locator, boolean asyncSolrC
     if (null != this.cfg.getBooleanQueryMaxClauseCount()) {
       IndexSearcher.setMaxClauseCount(this.cfg.getBooleanQueryMaxClauseCount());
     }
+    setWeakStringInterner();
```

Comment:
Please add a property to enable/disable this feature. Default on is fine, but ability to easily switch off for cases where it introduces issues would be nice.

Reply:
that's the plan.